contracts/Presale.sol

```diff
@@ -95,7 +109,7 @@ contract Presale is IPresale, Ownable2Step, ReentrancyGuard, Pausable {
     function setRounds(RoundConfig[] calldata newRounds) external override onlyOwner {
         emit RoundsUpdated($rounds, newRounds, _msgSender());
 
-        for (uint256 i; i < $rounds.length; ++i) {
+        while ($rounds.length != 0) {
```

Comment:
This way we don't constantly read from storage every iteration
```suggestion
        uint256 _totalRounds = $rounds.length;
        while (_totalRounds > 0) {
            _totalRounds--;
```

Reply:
added `_totalRounds` instead of loading all rounds into memory

Reply:
added `_totalRounds` instead of making storage read for length each iteration